src/storage/src/metrics/source/mysql.rs

```diff
@@ -7,21 +7,133 @@
 
-//! Metrics for Postgres.
+//! Metrics for MySQL.
 
-use mz_ore::metric;
-use mz_ore::metrics::{DeleteOnDropGauge, GaugeVec, GaugeVecExt, MetricsRegistry};
-use mz_repr::GlobalId;
-use prometheus::core::AtomicF64;
 use std::sync::Arc;
 use std::sync::Mutex;
 
+use prometheus::core::{AtomicF64, AtomicU64};
+
+use mz_ore::metric;
+use mz_ore::metrics::{
+    CounterVecExt, DeleteOnDropCounter, DeleteOnDropGauge, GaugeVec, GaugeVecExt, IntCounterVec,
+    MetricsRegistry, UIntGaugeVec,
+};
+use mz_repr::GlobalId;
+
 #[derive(Clone, Debug)]
 pub(crate) struct MySqlSourceMetricDefs {
-    pub(crate) table_count_latency: GaugeVec,
+    pub(crate) total_messages: IntCounterVec,
+    pub(crate) ignored_messages: IntCounterVec,
+    pub(crate) insert_rows: IntCounterVec,
+    pub(crate) update_rows: IntCounterVec,
+    pub(crate) delete_rows: IntCounterVec,
+    pub(crate) tables: UIntGaugeVec,
+    pub(crate) gtid: UIntGaugeVec,
+
+    pub(crate) snapshot_defs: MySqlSnapshotMetricDefs,
 }
 
 impl MySqlSourceMetricDefs {
+    pub(crate) fn register_with(registry: &MetricsRegistry) -> Self {
+        Self {
+            total_messages: registry.register(metric!(
+                name: "mz_mysql_per_source_messages_total",
+                help: "The total number of replication messages for this source, not expected to be the sum of the other values.",
+                var_labels: ["source_id"],
+            )),
+            ignored_messages: registry.register(metric!(
+                name: "mz_mysql_per_source_ignored_messages",
+                help: "The number of messages ignored because of an irrelevant type or relation_id",
+                var_labels: ["source_id"],
+            )),
+            insert_rows: registry.register(metric!(
+                name: "mz_mysql_per_source_inserts",
+                help: "The number of inserts for all tables in this source",
+                var_labels: ["source_id"],
+            )),
+            update_rows: registry.register(metric!(
+                name: "mz_mysql_per_source_updates",
+                help: "The number of updates for all tables in this source",
+                var_labels: ["source_id"],
+            )),
+            delete_rows: registry.register(metric!(
+                name: "mz_mysql_per_source_deletes",
+                help: "The number of deletes for all tables in this source",
+                var_labels: ["source_id"],
+            )),
+            tables: registry.register(metric!(
+                name: "mz_mysql_per_source_tables_count",
+                help: "The number of upstream tables for this source",
+                var_labels: ["source_id"],
+            )),
+            gtid: registry.register(metric!(
+                name: "mz_mysql_per_source_gtid",
+                help: "Transaction-id of the latest committed GTID for the specific GTID Source-ID UUID for this source",
+                var_labels: ["source_id", "gtid_source_uuid"],
+            )),
```

Comment:
This metric feels a bit weird since it is lossy when multiple uuids are involved and also we record the same info in the progress collection. No strong opinions, only wanted to note this thought

Reply:
That's a good point -- I wonder if it'd be better to report the cumulative transaction-ids of all GTID source-ids, so it's always an increasing value to show progress?

Reply:
I've updated this to use a single gauge for tracking the sum of all tx-ids from the full frontier